### ISSUE TITLE
Error making API request.

### ISSUE THREAD

**Issue by satheeshCharles** [hashicorp/vault#5718]
What could be the problem? Am i missing something ?

OS: centos

[root@salt vault]# vault secrets list
Path          Type         Accessor              Description
----          ----         --------              -----------
cubbyhole/    cubbyhole    cubbyhole_992aa5e9    per-token private secret storage
identity/     identity     identity_d809d242     identity store
kv/           kv           kv_b1ad68d4           n/a
secret/       kv           kv_006032b6           key/value secret storage
sys/          system       system_07b3cd6f       system endpoints used for control, policy and debugging
tia/          kv           kv_73f23057           n/a

[root@salt vault]# vault secrets enable secret
Error enabling: Error making API request.

URL: POST http://127.0.0.1:8200/v1/sys/mounts/secret
Code: 400. Errors:

* existing mount at secret/

**Comment by kalafut** [comment]
Note in the secrets list that a `kv` storage backend is already mounted (and therefore enabled) at the `secret/` path. Your command is trying to mount something else at that same location which is the source of the error.

Can you explain a bit about what you're trying to do?

**Comment by satheeshCharles** [comment]
Hi @kalafut  let me rephrase the question. I'm doing the POC.
the below paths are available in in the vault.

Path          Type         Accessor              Description
----          ----         --------              -----------
cubbyhole/    cubbyhole    cubbyhole_992aa5e9    per-token private secret storage
identity/     identity     identity_d809d242     identity store
kv/           kv           kv_f51d6f65           n/a
secret/       kv           kv_006032b6           key/value secret storage
sys/          system       system_07b3cd6f       system endpoints used for control, policy and debugging
testpath/     kv           kv_c7bcc7f0           n/a
tia/          kv           kv_73f23057           n/a

When I write the below is successful.

[root@salt log]# vault write kv/tia value="s3c(eT"
Success! Data written to: kv/tia

[root@salt log]# vault write system/sys value="s3c(eT"
Error writing data to system/sys: Error making API request.

URL: PUT http://127.0.0.1:8200/v1/system/sys
Code: 404. Errors:

* no handler for route 'system/sys'

Am I doing something wrong. ??

**Comment by kalafut** [comment]
Hi,

Your successful command `vault write kv/tia value="s3c(eT"` is saying write to  the "kv/tia" path the data "value=s3c(eT". This works because the backend at "kv/" is of type "kv" (key value). You would be able to do something similar to the other mounts  of type "kv", such as secret/, testpath/ and tia/. The backend at sys/ is of type "system", which is an entirely different set of capabilities. There is nothing mounted at "system/", hence the error.

Have you taken a look at our learning site? The guides there will walk through many of these concepts with a lot of good explanation: https://learn.hashicorp.com/vault/#getting-started

**Comment by satheeshCharles** [comment]
@kalafut Thanks.

**Comment by lakshmankashyap** [comment]
> vault init
> WARNING! The "vault init" command is deprecated. Please use "vault operator
> init" instead. This command will be removed in Vault 1.1.
> 
> Error initializing: Error making API request.
> 
> URL: PUT http://127.0.0.1:8200/v1/sys/init
> Code: 400. Errors:
> 
> * failed to check for initialization: Get http://127.0.0.1:8500/v1/kv/vault/core/?keys=&separator=%!F(MISSING): dial tcp 127.0.0.1:8500: connect: connection refused
>

**Comment by satheeshCharles** [comment]
Hi,

Seems the process in 8500 is not running , eventually it could be the
consul.

-Satheesh

**Comment by Alex-Mussell** [comment]
Hello @lakshmankashyap, did you manage to fix this error? How did you do it? Do you need to have a consul specific job?
